lib/MLDataDevices/ext/MLDataDevicesOpenCLExt.jl

```diff
@@ -0,0 +1,102 @@
+module MLDataDevicesOpenCLExt
+
+using Adapt: Adapt
+using GPUArrays: GPUArrays
+using MLDataDevices: MLDataDevices, Internal, OpenCLDevice, reset_gpu_device!
+using OpenCL: OpenCL, cl, CLArray
+
+const SUPPORTS_FP64 = Dict{cl.Device,Bool}()
+
+function __init__()
+    reset_gpu_device!()
+    for dev in vcat(cl.devices.(cl.platforms())...)
+        SUPPORTS_FP64[dev] = "cl_khr_fp64" in dev.extensions
+    end
+    return nothing
+end
+
+MLDataDevices.loaded(::Union{OpenCLDevice,Type{<:OpenCLDevice}}) = true
+function MLDataDevices.functional(::Union{OpenCLDevice,Type{<:OpenCLDevice}})
+    return try
+        cl.device()
+        true
+    catch
+        false
+    end
+end
```

Comment:
Is there no nicer way than a try/catch?

Reply:
This was the most straightforward way to test for functionality, but the change in the latest commit should work good enough.